### PR TITLE
Feat/yarn v2

### PR DESCRIPTION
**Pull request by gitphill**
- [X] Tests written and linted
- [x] Documentation written / README.md updated [https://snyk.io/docs/snyk-for-node/](i)
- [X] Follows [CONTRIBUTING agreement](CONTRIBUTING.md)
- [X] Commit history is tidy [https://git-scm.com/book/en/v2/Git-Branching-Rebasing](i)
- [ ] Reviewed by Snyk team

### What this does

Add support for yarn v2.
Metadata to inform Snyk CLI which version of yarn has been detected.
Refactor yarn 1 and 2 tests.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ dist
 .eslintcache
 *.log
 /package-lock.json
+.idea
```
Comment (dkontorovskyy) [inline .gitignore:9]: I still don't like this. None of our repos have this

Comment (gitphill) [reply .gitignore:9]: can take it out

Comment (dkontorovskyy) [reply .gitignore:9]: let's do that 🙏

**Comment by aviadhahami** [comment]
Rebased upon `master`

**Comment by lili2311** [comment]
https://github.com/snyk/nodejs-lockfile-parser/issues/56

**Pull request merged**